### PR TITLE
ServiceControl Distributed Audit Instances

### PR DESCRIPTION
**Pull request by boblangley**

### CODE REVIEW

File: servicecontrol/servicecontrol-instances/distributed-instances.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+title: Multiple ServiceControl Instances
+reviewed: 2017-07-10
+component: ServiceControl
+---
+
+### Known Constraints / Issues
+
+- Only supports Audits
+- SC Master instances must be configured with Address and API URI of remotes - JSON string
+- Paging for message view in SI is wonky
```
Comment (SeanFeldman) [inline servicecontrol/servicecontrol-instances/distributed-instances.md]: Is that what we want to write in the official doco? 🙂

**Comment by mikeminutillo** [comment]
```mermaid
graph TD
Endpoints -- Audits --> Audits(Audit Queue)
Endpoints -- Errors --> Errors(Error Queue)
Audits --> Master[ServiceControl<br/>Master]
Errors --> Master
Audits --> Slave[ServiceControl<br/>Slave]
Master -. API Calls .-> Slave
SI[ServiceInsight] -. API Calls .->Master
SP[ServicePulse] -. API Calls .-> Master
```

**Comment by SeanFeldman** [comment]
image

Looking at the diagram it's not obvious that audit queues would be different queues and not a single one.

**Comment by mikeminutillo** [comment]
They don't have to be different. We could add a 2nd audit queue to make it clearer. Or we could disconnect the master from the audits too. This wasn't meant to be definitive, it was just a diagram I had slapped together the other day.

**Comment by SzymonPobiega** [comment]
I would disconnect the master from the audit in this diagram. Or we can draw multiple diagrams for various use cases:
 * Scaling out from scratch
 * Adding a master to handle recoverability while retaining an existing instance for audits
 * Multi-region handling

**Comment by SzymonPobiega** [comment]
@mikeminutillo @danielmarbach I can work on this piece of doco today. I am not feeling well and would like to take it easy.

File: servicecontrol/servicecontrol-instances/distributed-instances.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+---
+title: Multiple ServiceControl Instances
+reviewed: 2017-07-16
+component: ServiceControl
+---
+
+### Known Constraints / Issues
+
+- Only supports Audits
+- SC Master instances must be configured with Address and API URI of remotes - JSON string
```
Comment (SeanFeldman) [inline servicecontrol/servicecontrol-instances/distributed-instances.md]: We should provide a sample.

File: servicecontrol/servicecontrol-instances/distributed-instances.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+- Paging for message view in SI is wonky
+- Data from remote instances that cannot be reached by the master instance will not be included in the results
```
Comment (SeanFeldman) [inline servicecontrol/servicecontrol-instances/distributed-instances.md]: What results? This is understandable to us who worked on the feature, but not a general userr.

File: servicecontrol/servicecontrol-instances/distributed-instances.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+### Disabling Recoverability
+
+- Set Error queue name to `!disabled`
```
Comment (SeanFeldman) [inline servicecontrol/servicecontrol-instances/distributed-instances.md]: Would linking this to the setting doco be helpful?

**Comment by danielmarbach** [comment]
Rebased for now

**Comment by danielmarbach** [comment]
Like it!

File: servicecontrol/servicecontrol-instances/distributed-instances.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  <appSettings>
+    <add key="ServiceControl/RemoteInstances" value="[{'api_uri':'http://localhost:33334/api', 'queue_address':'Particular.ServiceControl.Slave'}]'"/>
+  </appSettings>/
+</configuration>
+```
+
+4. Start ServiceControl master.
+5. Validate ServiceInsight and ServicePulse are connecting to the master instance only.
+
+### Splitting an existing installation
```
Comment (WilliamBZA) [inline servicecontrol/servicecontrol-instances/distributed-instances.md]: This section needs quite a bit more fleshing out

Comment (danielmarbach) [reply servicecontrol/servicecontrol-instances/distributed-instances.md]: Can you be a bit more specific? 😀

Comment (SeanFeldman) [reply servicecontrol/servicecontrol-instances/distributed-instances.md]: Suggest to merge the PR and "polish" it later if needed. Would that work?

**Comment by boblangley** [comment]
I agree. I think we can stand to get more feedback from customers before doing more polishing. I am going to merge this for now.

**Pull request merged**